airflow/executors/executor_loader.py

```diff
@@ -31,47 +32,108 @@ class ExecutorLoader:
-    def _get_executor(executor_name: str) -> BaseExecutor:
+    def create_sequential_executor() -> BaseExecutor:
+        """Creates SequentialExecutor"""
+        from airflow.executors.sequential_executor import SequentialExecutor
+        return SequentialExecutor()
+
+    @staticmethod
+    def create_celery_executor() -> BaseExecutor:
+        """Creates CeleryExecutor"""
+        from airflow.executors.celery_executor import CeleryExecutor
+        return CeleryExecutor()
+
+    @staticmethod
+    def create_dask_executor() -> BaseExecutor:
+        """Creates DaskExecutor"""
+        from airflow.executors.dask_executor import DaskExecutor
+        return DaskExecutor()
+
+    @staticmethod
+    def create_kubernetes_executor() -> BaseExecutor:
+        """Creates KubernetesExecutor"""
+        from airflow.executors.kubernetes_executor import KubernetesExecutor
+        return KubernetesExecutor()
+
+    @classmethod
+    def _get_executor(cls, executor_name: str) -> BaseExecutor:
         """
         Creates a new instance of the named executor.
         In case the executor name is unknown in airflow,
-        look for it in the plugins
+        look for it in the plugins.
         """
         if executor_name == ExecutorLoader.LOCAL_EXECUTOR:
-            from airflow.executors.local_executor import LocalExecutor
-            return LocalExecutor()
+            return cls.get_or_create_executor(executor_name=executor_name,
+                                              create_executor=ExecutorLoader.create_local_executor)
         elif executor_name == ExecutorLoader.SEQUENTIAL_EXECUTOR:
-            from airflow.executors.sequential_executor import SequentialExecutor
-            return SequentialExecutor()
+            return cls.get_or_create_executor(executor_name=executor_name,
+                                              create_executor=ExecutorLoader.create_sequential_executor)
         elif executor_name == ExecutorLoader.CELERY_EXECUTOR:
-            from airflow.executors.celery_executor import CeleryExecutor
-            return CeleryExecutor()
+            return cls.get_or_create_executor(executor_name=executor_name,
+                                              create_executor=ExecutorLoader.create_celery_executor)
         elif executor_name == ExecutorLoader.DASK_EXECUTOR:
-            from airflow.executors.dask_executor import DaskExecutor
-            return DaskExecutor()
+            return cls.get_or_create_executor(executor_name=executor_name,
+                                              create_executor=ExecutorLoader.create_dask_executor)
         elif executor_name == ExecutorLoader.KUBERNETES_EXECUTOR:
-            from airflow.executors.kubernetes_executor import KubernetesExecutor
-            return KubernetesExecutor()
+            return cls.get_or_create_executor(executor_name=executor_name,
+                                              create_executor=ExecutorLoader.create_kubernetes_executor)
```

Comment:
I wonder if we can use some mapping here because pylint is going to complain about to much return statements when we add one more executor.

```python
def _get_executor(cls, executor_name: str) -> BaseExecutor:
    mapping = {
        ExecutorLoader.LOCAL_EXECUTOR: ExecutorLoader.create_local_executor
        ...
    }
    if executor_name in mapping:
        method = mapping[executor_name]
        return cls.get_or_create_executor(executor_name, method)
    else:
        ...
```

Reply:
sure. We can set variable and make single return :)